libsolidity/parsing/Parser.cpp

```diff
@@ -1313,11 +1314,32 @@ ASTPointer<InlineAssembly> Parser::parseInlineAssembly(ASTPointer<ASTString> con
 
 	expectToken(Token::Assembly);
 	yul::Dialect const& dialect = yul::EVMDialect::strictAssemblyForEVM(m_evmVersion);
+	ASTPointer<vector<ASTPointer<ASTString>>> dialectFlags;
 	if (m_scanner->currentToken() == Token::StringLiteral)
 	{
-		if (m_scanner->currentLiteral() != "evmasm")
-			fatalParserError(4531_error, "Only \"evmasm\" supported.");
+		std::string const& dialectString = m_scanner->currentLiteral();
+		auto it = dialectString.begin();
```

Comment:
Why don't you just use `boost::split`?

Reply:
Is that locale-whitespace-safe and sane and everything :-)? I was actually wondering if we had our own custom helper for this, but didn't find one...

Reply:
If is locale-aware only if you use locale-aware separators.